sklearn/preprocessing/_function_transformer.py

```diff
@@ -89,6 +91,20 @@ def __init__(self, func=None, inverse_func=None, validate=None,
         self.kw_args = kw_args
         self.inv_kw_args = inv_kw_args
 
+    def __getstate__(self):
+        self.func_name = self.func.__name__
+        self.func_code = marshal.dumps(self.func.__code__)
```

Comment:
This way of serializing user-defined function from the `__main__` module is incomplete: several attributes of the function are not extracted. Doing it properly is not trivial, and that is partly the purpose of [cloudpickle](https://github.com/cloudpipe/cloudpickle).

Reply:
`cloudpickle` is vendored inside `joblib` which is itself vendored inside `scikit-learn` (arguably hard to find).

Reply:
(IIRC the plan is to stop vendoring joblib in the future and make it a dependency of scikit-learn)